test/e2e/dead_start_test.go

```diff
@@ -0,0 +1,296 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgtest "knative.dev/pkg/test"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	"knative.dev/serving/pkg/apis/serving"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	v1options "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+)
+
+// withConfigImage sets the container image to be the provided string.
+func withConfigImage(img string) v1options.ServiceOption {
+	return func(svc *v1.Service) {
+		svc.Spec.Template.Spec.PodSpec.Containers[0].Image = img
+	}
+}
+
+func generationLabelSelector(config string, generation int) string {
+	return fmt.Sprintf("%s=%s,%s=%s", serving.ConfigurationLabelKey, config, "serving.knative.dev/configurationGeneration", strconv.Itoa(generation))
+}
+
+// This test case creates a service which can never reach a ready state.
+// The service is then udpated with a healthy image and is verified that
+// the healthy revision is ready and the unhealhy revision is scaled to zero.
+func TestDeadStartToHealthy(t *testing.T) {
```

Comment:
I added a similar test here - https://github.com/knative/serving/pull/14909

But I was expecting it to fail without any fixes - but it actually did pass. So then I realized that _we_ already do scale down crashing revisions when they are unreachable.

Reply:
The case here is still happening for me. https://github.com/knative/serving/issues/14656#issuecomment-1823493546
The rev-2 will scale down but the first stuck in restarting.